### DISCUSSION TITLE
vscode plugin?

### DISCUSSION THREAD

**Discussion by pm100** [dotenv-linter/dotenv-linter#764] [Ideas]
I know almost zero about vscode plugins but a .env plugin using the validator would be cool.

At least a plugin that runs checks on save

**Comment by DDtKey** [comment]
Neat idea, it also can be integrated with `JetBrains`.

But first of all, I think this will require support for a machine-readable output format. To make integrations possible

**Reply by mgzenitech** [reply]
There is a way sorta to do it through any-lint. I haven't managed to completely make it work but my progress can be found here if anyone's interested: 
https://github.com/tamayika/vscode-any-lint/issues/12

**Reply by ccoVeille** [reply]
Thanks for sharing, I will try using it with ls-lint. 

https://ls-lint.org for reference

**Reply by mgzenitech** [reply]
that's it working with any-lint already :) plugin author fixed the issue.

**Comment by mgzenitech** [comment]
![Screenshot from 2024-07-15 09-39-31](https://github.com/user-attachments/assets/8eafd24e-f338-4081-bcf0-53c3bd817597)

**Comment by mgzenitech** [comment]
The only issue here is that `dotenv-linter` should support `--stdin` to be able to work with `onType` event and not only `onSave`.

More info @ https://github.com/tamayika/vscode-any-lint/issues/1

So if any devs of this linter will add these (or similar) switches to cli:
```
--stdin
--stdin-filename
```
then any-lint will also be able to provide onType event info

**Comment by mgzenitech** [comment]
For documentation purposes will leave also config here to run dotenv-linter through any-lint:
```json
"any-lint.linters": [
    {
      "args": [
        "-q",
        "${file}"
      ],
      "binPath": "dotenv-linter",
      "condition": "$.fileExtname.endsWith('.env')",
      "cwd": "${workspaceFolder}",
      "diagnostic": {
        "format": "${file}:${startLine} ${message}",
        "output": "stdout"
      },
      "name": "dotenv-linter",
      "on": [
        "change",
        "save"
      ]
    }
  ]
```

**Comment by mgzenitech** [comment]
Here you go guys, I've made example config that adds full support for dotenv-linter via any-linter:
https://github.com/tamayika/vscode-any-lint/pull/26/files

For now only onsave event works as dotenv-linter has to [add support for --stdin ](https://github.com/dotenv-linter/dotenv-linter/issues/803)
